packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js

```diff
@@ -0,0 +1,189 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { parseFixed } = require("@ethersproject/bignumber");
+
+// An implementation of PriceFeedInterface that takes as input two sets ("baskets") of price feeds,
+// computes the average price feed for each basket, and returns the spread between the two averages.
+// !!Note: This PriceFeed assumes that the baselinePriceFeeds, experimentalPriceFeed, and denominatorPriceFeed
+// are all returning prices in the same precision as `decimals`.
+class BasketSpreadPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new BasketSpreadPriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {List} baselinePriceFeeds The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {List} experimentalPriceFeed The baseline list of priceFeeds to compute the average of. All elements must be of type PriceFeedInterface.
+   *      Must be an array of at least one element.
+   * @param {Object} denominatorPriceFeed We divide the price spread between the baseline and experimental baskets by this denominator price
+   *      in order to "denominate" the basket spread price in a specified unit. For example, we might want to express the basket spread in terms
+   *      of ETH-USD.
+   */
+  constructor(web3, logger, baselinePriceFeeds, experimentalPriceFeeds, denominatorPriceFeed) {
+    super();
+
+    if (baselinePriceFeeds.length === 0 || experimentalPriceFeeds.length === 0) {
+      throw new Error("BasketSpreadPriceFeed cannot be constructed with empty baseline or experimental baskets.");
+    }
+
+    this.baselinePriceFeeds = baselinePriceFeeds;
+    this.experimentalPriceFeeds = experimentalPriceFeeds;
+    this.denominatorPriceFeed = denominatorPriceFeed;
```

Comment:
How might one parameterize this denominator feed for something like USDC?

Reply:
Yeah so this is an open question. We could allow denominatorPriceFeed to be `null` in which case we just ignore the last computation step.

Reply:
Done PTAL

Reply:
+1 -- that seems simpler